### ISSUE TITLE
learning policy

### ISSUE THREAD

**Issue by AtefehBaniasadi** [open-mmlab/mmocr#681]
Hello

What is the learning policy on text detection models and what is set and with what values?

Thanks in advance,

**Comment by cuhk-hbsun** [comment]
Please check the `optimizer` setting under [folder](https://github.com/open-mmlab/mmocr/tree/main/configs/_base_/schedules).
For example, for [panet_r18_fpem_ffm_600e_ctw1500.py](https://github.com/open-mmlab/mmocr/blob/main/configs/textdet/panet/panet_r18_fpem_ffm_600e_ctw1500.py#L2), the learning policy is in [schedule](https://github.com/open-mmlab/mmocr/blob/main/configs/_base_/schedules/schedule_adam_600e.py)

**Comment by AtefehBaniasadi** [comment]
Is there any relation between the type of learning policy and the number of points for each bounding box for a text, to be more clear I have just used 4 points for each text bounding box. Is it right to select   'poly' as a learning policy?

I thought for policy='poly' i must use more than 4 points to draw a bounding box for each text
can you please explain it to be more clear to me?

Thanks,

**Comment by gaotongxiao** [comment]
Learning policy has nothing to do with the type of bounding box, they just happen to share the same name. Learning policy only determines how the learning rate changes over time, and passing 'poly' to `lr_config' will call [PolyLrUpdaterHook](https://github.com/open-mmlab/mmcv/blob/e1d6ea8c8706a8515782a026bde8e845930640da/mmcv/runner/hooks/lr_updater.py#L226) to apply its policy.
https://github.com/open-mmlab/mmocr/blob/31f1c007c0c294ea5398a87b342b1fd18c29ee4e/configs/_base_/schedules/schedule_adam_600e.py#L5

Your confusion might have arisen from the `text_repr_type` in detection models' postprocessors. Here you can switch between `poly` and `quad` based on the number of points for each bounding box and it affects how the model outputs are processed into boundary boxes.
https://github.com/open-mmlab/mmocr/blob/fb1892a1aebc930836c1da8588bbc645b879e41f/configs/_base_/det_models/textsnake_r50_fpn_unet.py#L20

BTW, we're writing a config tutorial and beginners are expected to get the most out of it after reading it. If you are interested, you can [preview our tutorial](https://mmocr--683.org.readthedocs.build/en/683/tutorials/config.html) now and share your thoughts with us if any ;)

**Comment by AtefehBaniasadi** [comment]
> Learning policy has nothing to do with the type of bounding box, they just happen to share the same name. Learning policy only determines how the learning rate changes over time, and passing 'poly' to `lr_config' will call [PolyLrUpdaterHook](https://github.com/open-mmlab/mmcv/blob/e1d6ea8c8706a8515782a026bde8e845930640da/mmcv/runner/hooks/lr_updater.py#L226) to apply its policy.
> 
> https://github.com/open-mmlab/mmocr/blob/31f1c007c0c294ea5398a87b342b1fd18c29ee4e/configs/_base_/schedules/schedule_adam_600e.py#L5
> 
> Your confusion might have arisen from the `text_repr_type` in detection models' postprocessors. Here you can switch between `poly` and `quad` based on the number of points for each bounding box and it affects how the model outputs are processed into boundary boxes.
> 
> https://github.com/open-mmlab/mmocr/blob/fb1892a1aebc930836c1da8588bbc645b879e41f/configs/_base_/det_models/textsnake_r50_fpn_unet.py#L20
> 
> BTW, we're writing a config tutorial and beginners are expected to get the most out of it after reading it. If you are interested, you can [preview our tutorial](https://mmocr--683.org.readthedocs.build/en/683/tutorials/config.html) now and share your thoughts with us if any ;)

Thank you for your attention,
I definitely read your tutorial and send my feedback to you.

**Comment by AtefehBaniasadi** [comment]
Hello Dear,

I have trained the Snake model on my custom dataset but the 'loss_text' didn't decrease more than 0.30.
 what is wrong with that my model can not converge? is there any specific config to converge rapidly?

my configs are as bellow:
```
# optimizer
optimizer = dict(type='Adam', lr=0.00005, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
# learning policy
lr_config = dict(policy='poly', power=0.9, min_lr=1e-7, by_epoch=True)
total_epochs = 700

I have trained my model on a single GPU and I set  samples_per_gpu (batch size) with 8
```

Many thanks for your attention

**Comment by gaotongxiao** [comment]
@AtefehBaniasadi Do you mean the TextSnake model? Your learning rate may be too small for the model to converge. According to the [metafile](https://github.com/open-mmlab/mmocr/blob/dab5725d70b92ada4383565c3e60f6634616fd60/configs/textdet/textsnake/metafile.yml), TextSnake was trained on 8 GPUs with the actual batch size 32 (samples_per_gpus * num_gpus). Even if you want to scale down the learning rate in proportion to how your actual batch size change, the learning rate should not be less than 0.007/(32 / 8)=0.00175.
